praw/models/reddit/comment.py

```diff
@@ -1,12 +1,15 @@
 """Provide the Comment class."""
+from six import string_types
+
 from ...exceptions import ClientException
 from ..comment_forest import CommentForest
 from .base import RedditBase
 from .mixins import InboxableMixin, ThingModerationMixin, UserContentMixin
 from .redditor import Redditor
+from .subreddit import Subreddit
 
 
-class Comment(RedditBase, InboxableMixin, UserContentMixin):
+class Comment(InboxableMixin, UserContentMixin, RedditBase):
```

Comment:
While I'm sure you are correct, could you please provide a reference for that? Thanks!

Reply:
Since mixins are just a way of organising code, member lookups on classes should consult the members of its mixins before looking at the members of its base class. This is why bases should be written on the right side.

I saw a blog post on this before. I’ve found a few now:

[Mixins and Python](https://www.ianlewis.org/en/mixins-and-python):

> Mixins are a really great concept but I often find that people use them incorrectly which can lead to some bugs.

[Python Base Class Order](https://www.georgevreilly.com/blog/2016/01/14/PythonBaseClassOrder.html):

> To get the desired behavior of the mixin overriding the base, `Base` should always appear *last* in the in­her­i­tance list.

[Multiple inheritance is hard](https://nedbatchelder.com/blog/201210/multiple_inheritance_is_hard.html):

> Mixins come first even though it looks odd.